app/services/hyku_addons/task_master/publish_service.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require "google/cloud/pubsub"
+
+module HykuAddons
+  module TaskMaster
+    class PublishService
+      ALLOWED_TYPES = %w[tenant file work].freeze
+      ALLOWED_ACTIONS = %w[create update destroy].freeze
+
+      ##
+      # @api public
+      #
+      # Init
+
+      # @param [String] the type instance type
+      # @param [String] the CRUD action being performed
+      # @param [JSON] the JSON string to be published
+      #
+      # @return [HykuAddons::TaskMaster::PublishService]
+      def initialize(type, action, json)
+        @type = type
+        @action = action
+        @json = json
+
+        validate_arguments!
+      end
+
+      ##
+      # @api public
+      #
+      # Publish a message to Google PubSub
+      #
+      # @return [Google::Cloud::PubSub::Message] or nil
+      def perform
+        topic = client.topic(topic_name)
+        topic.publish(@json)
+      end
+
+      protected
+
+        # repository--<type>-<action>: repository--work-create
+        def topic_name
+          "repository--#{@type}-#{@action}"
+        end
+
+        def client
+          @client ||= begin
+                        Google::Cloud::PubSub.configure do |config|
+                          config.project_id  = pubsub_credentials.dig("project_id")
+                          config.credentials = pubsub_credentials
+                        end
+
+                        Google::Cloud::PubSub.new
+                      end
+        end
+
+        def pubsub_credentials
+          raise KeyError, "Service environment variable is not set" unless ENV["PUBSUB_SERVICEACCOUNT_KEY"].present?
+
+          @pubsub_credentials ||= JSON.parse(ENV["PUBSUB_SERVICEACCOUNT_KEY"])
```

Comment:
I wonder if we should pipe this through `Settings` as well.  This isn't vital right now just think it might be good to do eventually.